backend/src/user/endpoints.ts

```diff
@@ -70,7 +71,7 @@ export const authLogout = (req: Request, res: Response) => {
 
 export const authVerify = (req: Request, res: Response) => {
     if (req.isAuthenticated()) {
-        res.status(200).json({ loggedIn: true, user: (req.user as RequestData).user });
+        res.status(200).json({ loggedIn: true, user: req.user as RequestData });
```

Comment:
Tutaj za mało usunąłeś ponieważ to `as RequestData` jest niepotrzebne, a nawet niepoprawne, bo `req.user` to `User`, a nie `RequestData`